tutorials/3d/procedural_geometry/arraymesh.rst

```diff
@@ -126,6 +142,24 @@ by adding each array to ``surface_array`` and then committing to the mesh.
 
     mesh.add_surface_from_arrays(Mesh.PRIMITIVE_TRIANGLES, surface_array) # No blendshapes, lods, or compression used.
 
+ .. code-tab:: csharp C#
+
+    var verts = vertList.ToArray();
```

Comment:
Do we need to store references to the intermediate arrays here?  

Also, my default would be to use a `Variant[]` until we have to pass it to `AddSurfaceFromArrays`. Would make less engine calls. Maybe @raulsntos can agree/disagree?

```cs
Variant[] surfaceArray = new Variant[(int)Mesh.ArrayType.Max];

surfaceArray[(int)Mesh.ArrayType.Vertex] = vertsList.ToArray();
// ...

arrMesh.AddSurfaceFromArrays(Mesh.PrimitiveType.Triangles, new Godot.Collections.Array(surfaceArray));
```

Reply:
> Do we need to store references to the intermediate arrays here?

No, I don't think so. Fixed and renamed the List<> variables to match the GDScript examples.

> Also, my default would be to use a Variant[] until we have to pass it to AddSurfaceFromArrays. Would make less engine calls.

That's well beyond my knowledge of the bindings, so I'm learning something here. It might be more consistent with other docs to use the Array, since the [API differences page](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_differences.html) lists Godot.Collections.Array as the equivalent type to PackedVector\*Array, but that might be a secondary concern. I'm out of my depth on this question, so I'm happy to wait for further comment and/or be guided by those who know more than I do.

Reply:
@paulloz I don't think it would result in less engine calls, the `Godot.Collections.Array` constructor from `Variant[]` ends up doing the same thing that the current example is doing but with an extra `Variant[]` allocation:

https://github.com/godotengine/godot/blob/f2aae8fa5c2e9d9323832fb43c8446c2e518d698/modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs#L55-L69

Oh wait, I think I get what you mean. So you are saying it would be better to avoid constructing the `Godot.Collections.Array` in case we don't enter the `if (arrMesh != null)`? I agree it would be better, but I'm not sure if such optimizations should be in the documentation where the focus is more on readability.

Reply:
@raulsntos I think I got confused. I thought we were making a more efficient memory copy here, but it's only when dealing with `Packed*Array` on the engine side. Sorry, don't mind me then. It's perfectly fine how it is.